### ISSUE TITLE
no-undef doesn't understand commonjs module scope

### ISSUE THREAD

**Issue by jkroso** [eslint/eslint#892]
I think it would be nice to turn this warning off when I know I'm working in a commonjs module.

```js
var toString = 1 // => 'toString' is read only
```

**Comment by feross** [comment]
Any progress getting this into the parser?

**Comment by nzakas** [comment]
No. This issue will be updated when something changes.

**Comment by feross** [comment]
Esprima bug filed: https://code.google.com/p/esprima/issues/detail?id=630&thanks=630&ts=1422662287

Please star it.

**Comment by nzakas** [comment]
We're not using Esprima, we use Espree (https://github.com/eslint/espree).

**Comment by feross** [comment]
Oh, sorry for the confusion.

**Comment by nzakas** [comment]
Convinced escope to add a special scope for Node :) Working on this.
